src/box/lua/console.lua

```diff
@@ -649,19 +670,34 @@ local function local_read(self)
             break
         end
         if delim == "" then
-            local lang = self.language
-            if not lang or lang == 'lua' then
-                -- stop once a complete Lua statement is entered
-                if local_check_lua(buf) then
-                    break
-                end
+            local continuation_on
+            if self.remote ~= nil then
+                continuation_on = self.remote.continuation_on
             else
-                break
+                continuation_on = self.continuation_on
+            end
+            local lang = self.language or 'lua'
+            -- Continue reading if Lua input seems incomplete according to
+            -- internal heuristic.
+            if lang == 'lua' and not local_check_lua(buf) then
+                goto continue
+            end
+            -- Interpret continuation_symbol as a line continuation marker.
+            --
+            -- This check is performed after local_check_lua() because in some
+            -- rare cases (e.g. in multiline strings with line ending with
+            -- slash) the slash would be treated as continuation marker and
+            -- removed.
```

Comment:
I said that I don't mind it and I still don't mind, but I would highlight that AFAIS it is inconsistent with the remote console. Need to re-check with Igor I think.

Reply:
What does seem inconsistent to you? The same `local_read()` used with the same results:
```
Tarantool 2.10.0-beta1-1838-g55e9607eb
type 'help' for interactive help
tarantool> require("console").connect("127.0.0.1:53493")
connected to 127.0.0.1:53493
---
- true
...

127.0.0.1:53493> print([[
               > aaa \
               > bbb
               > ]])
---
...

127.0.0.1:53493> return ([[
aaa \
bbb
]])
---
- 'aaa \

  bbb

  '
...

```
On server the right string is being printed:
```
tarantool> require("console").listen("127.0.0.1:53493")
started
---
- fd: 12
  name:
    host: 127.0.0.1
    family: AF_INET
    type: SOCK_STREAM
    protocol: tcp
    port: 53493
...

aaa \
bbb
```

Reply:
Thanks for the details! It means that my guess was wrong.

Reply:
(I did not forget to turn continuation on)